java/org/apache/catalina/realm/CombinedRealm.java

```diff
@@ -386,6 +387,38 @@ public Principal authenticate(GSSContext gssContext, boolean storeCred) {
         return null;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Principal authenticate(GSSName gssName, GSSCredential gssCredential) {
+        Principal authenticatedUser = null;
+        String username = gssName.toString();
```

Comment:
This looks to be unnecessary. You should be able to use gssName directly in the logging calls. This could be cleaned up in the existing code too.

Reply:
That's true, I have simply reused the current code pattern from others methods to all look alike. This can be cleaned up in `RealmBase` is a distinct subsequent commit.